src/org/labkey/test/pages/query/EditMetadataPage.java

```diff
@@ -0,0 +1,88 @@
+import org.openqa.selenium.WebElement;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+
+public class EditMetadataPage extends BaseDomainDesigner<EditMetadataPage.ElementCache>
+{
+    public EditMetadataPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+
+    public DataRegionTable clickViewData(BaseWebDriverTest test, String schemaName, String queryName)
+    {
+        return test.viewQueryData(schemaName, queryName);
+    }
+
+    public void clickEditSourcePage()
+    {
+        elementCache().editSourceBtn.click();
+    }
+
+    public EditMetadataPage clickSave()
+    {
+        elementCache().saveBtn.click();
+        return this;
+    }
+
+    public void clickResetToDefault()
+    {
+        WebDriverWrapper.waitFor(elementCache().resetToDefaultBtn::isEnabled,
+                "The alias button did not become enabled as expected", 2000);
+        elementCache().resetToDefaultBtn.click();
+        new ModalDialog(new ModalDialog.ModalDialogFinder(getDriver()).withTitle("Confirm Reset")).dismiss("Reset");
+    }
+
+    public EditMetadataPage clickAliasField(String fieldName)
+    {
+        WebDriverWrapper.waitFor(elementCache().aliasFieldBtn::isEnabled,
+                "The alias button did not become enabled as expected", 2000);
+        elementCache().aliasFieldBtn.click();
+        ModalDialog dialog = new ModalDialog(new ModalDialog.ModalDialogFinder(getDriver()).withTitle("Choose a field to wrap"));
+        ReactSelect.finder(getDriver()).findWhenNeeded(dialog).select(fieldName);
+        dialog.dismiss("OK");
+
+        return this;
+    }
+
+    public DomainFormPanel fieldsPanel()
+    {
+        return elementCache().firstDomainFormPanel;
+    }
+
+    @Override
+    protected EditMetadataPage.ElementCache newElementCache()
+    {
+        return new EditMetadataPage.ElementCache();
+    }
+
+    protected class ElementCache extends BaseDomainDesigner<EditMetadataPage.ElementCache>.ElementCache
+    {
+        protected final WebElement buttonPanel = queryMetadataButtonPanelLocator().findWhenNeeded(this);
+        protected final WebElement aliasFieldBtn = Locator.button("Alias Field").findWhenNeeded(buttonPanel);
+        protected final WebElement editSourceBtn = Locator.button("Edit Source").findWhenNeeded(buttonPanel);
+        protected final WebElement saveBtn = Locator.button("Save").findWhenNeeded(getDriver());
+        protected final WebElement resetToDefaultBtn = Locator.button("Reset To Default").findWhenNeeded(buttonPanel);
+        DomainFormPanel firstDomainFormPanel = new DomainFormPanel.DomainFormPanelFinder(getDriver())   // for situations where there's only one on the page
+                .timeout(WAIT_FOR_JAVASCRIPT)
+                .findWhenNeeded(this);
+
+        protected Locator.XPathLocator queryMetadataButtonPanelLocator()
+        {
+            return Locator.byClass("query-metadata-editor-buttons");
+        }
+
+    }
+}
```

Comment:
There is nothing wrong with having the macro like functions you have created for this page (ResetToDefault, AliasField) that choose the default action for any confirmation dialogs that appear. However keep in mind it does limit the kind of tests that can use the form. For example if there is a test that wants to validate that canceling out of a dialog works as expected, or one that expects an error message when updating, they will not be able to use the methods already defined.

This approach may be perfectly ok for this feature/test but it is something to keep in mind for the future.

Reply:
ok